app/models/host/discovered.rb

```diff
@@ -70,9 +70,9 @@ def attributes_to_import_from_facts
     super
   end
 
-  def populate_fields_from_facts(facts = self.facts_hash, type = 'puppet')
+  def populate_fields_from_facts(facts = self.facts_hash, ignored_type)
     # detect interfaces and primary interface using extensions
-    parser = super(facts, type)
+    parser = super(facts, :foreman_discovery)
```

Comment:
We explicitly set the parser here, therefore it will only accept discovered hosts. I will raise an error if the host is not discovered one.